File: CodingTest/프로그래머스/Level_1/python/체육복.py
def solution(n, lost, reserve):
    answer = 0
    
    lost_check = [1] * (n + 1)
    
    for i in lost:
        lost_check[i] = 0
        if reserve.count(i) == 1:
            reserve.remove(i)
            lost_check[i] = 1
        
    for i in range(1, n + 1):
        if lost_check[i] == 0:
            if  1 < i and reserve.count(i - 1) == 1:
                lost_check[i] = 1
                reserve.remove(i - 1)
            elif i < n and reserve.count(i + 1) == 1:
                lost_check[i] = 1
                reserve.remove(i + 1)
                
    answer = sum(lost_check) - 1
        
    return answer

File: CodingTest/프로그래머스/Level_1/swift/체육복.swift
func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
  var arr = [0] + Array(repeating: 1, count: n) + [0]

  let reserves = reserve.filter{!lost.contains($0)}
  let losts = lost.filter{!reserve.contains($0)}
  reserves.forEach{ arr[$0] = 2 }
  losts.forEach{ arr[$0] = 0 }
    
  for i in 1...n {
    if arr[i] == 0{
      if arr[i - 1] == 2{
        arr[i] = 1
        arr[i - 1] = 1
      } else if arr[i + 1] == 2{
        arr[i] = 1
        arr[i + 1] = 1
      }
    }
  }
  
  
  return arr.filter{$0 == 1 || $0 == 2}.count
}